app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -4,7 +4,7 @@ class Api::V2::ActivationKeysController < Api::V2::ApiController
 
     before_action :verify_presence_of_organization_or_environment, :only => [:index]
     before_action :find_environment, :only => [:index, :create, :update]
-    before_action :find_optional_organization, :only => [:index, :create, :show]
+    before_action :find_optional_nested_object, :only => [:index, :create, :show]
```

Comment:
I'll wrap it and fix tests. According to the failures, I think a lot of of the permissions were not properly enforced before using the methods from core.